### PR TITLE
fix #3756 on linux and make output closer to gnu date on other oses

### LINKED ISSUE #3756: `date` behaves differently with no args
`date` behaves differently than coreutils when no argument is passed,  if the goal of the project is to behave a 100% like the gnu coreutils, this should be fixed. here are two examples , it should be pretty easy to reproduce unless it's a weird problem with both my systems (manjaro and endeavour os)

---

Exemple 1 :

image
`$./date`
`Fri Jul 29 09:09:10 2022 `
`$date`
`$Fri 29 Jul 2022 09:09:12 CEST`

---

Exemple 2 :

image
`$./date`
`Fri Jul 29 16:40:39 2022 `
`$date`
`$Fri Jul 29 04:40:40 PM CDT 2022`

---

I looked at the code and i think it's caused by the chrono dependency not correctly detecting the current locale when formatting with %c. not sure about that though.

### PR DESCRIPTION
**Pull request by rayan-mazouz**
makes `date` output identical to `date` from the gnu core utils on Linux (as far as I tested).

output on windows and mac is still not identical
redox not tested

additional crates :
https://crates.io/crates/libc-strftime (MIT license)

tested on Linux, Mac os and Windows

ps : I accidentally pulled  a commit form the main branch, but I didn't modify anything about `ls`

### CODE REVIEW

File: src/uu/date/src/date.rs [hunk 1]
```diff
@@ -55,6 +59,9 @@ const OPT_REFERENCE: &str = "reference";
 const OPT_UNIVERSAL: &str = "universal";
 const OPT_UNIVERSAL_2: &str = "utc";
 
+#[cfg(all(unix, not(target_os = "macos"), not(target_os = "redox")))]
```
Comment (sylvestre) [inline src/uu/date/src/date.rs:62]: please add a comment explaining what this number is about?
thanks

**Comment by tertsdiepraam** [comment]
This feels like a temporary (but good!) solution. What would need to happen to solve this permanently? Are there issues we can reference in the code so we can keep track of the progress of fixing this in chrono for example?

**Comment by rayan-mazouz** [comment]
i agree this is definitely not an ideal solution, but at least it works for now. 
here's a list (from memory, so probably non exhaustive) of things that would need to be fixed
 in chrono in order to have a more elegant solution :
- fixing locale-dependent formatting specifiers like "%c" and "%X" in `chrono::format()` (see https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html).  issue : https://github.com/chronotope/chrono/issues/708

- fix the "%Z" specifier (timezone) doesn't behave like expected in `chrono::format()` : prints offset from the local timezone and UTC instead of the name of the timezone. issue : https://github.com/chronotope/chrono/issues/749

- `chrono::format_localized()` (enabled with chrono's `unstable-locales` feature) doesn't auto detect locale, and doesn't allow specifying a locale as a String, only a `chrono::Locale` and i couldn't find any way to get a `chrono::Locale` from a String (there's no issue as it's intended, although limiting behavior)

I'll reference both issues in the code

**Comment by tertsdiepraam** [comment]
Thanks!

**Comment by sylvestre** [comment]
I guess you saw:
```

---- test_date::test_date_format_day stdout ----
current_directory_resolved: 
run: D:\a\coreutils\coreutils\target\debug\coreutils.exe date +%a
run: D:\a\coreutils\coreutils\target\debug\coreutils.exe date +%A
run: D:\a\coreutils\coreutils\target\debug\coreutils.exe date +%u
---- test_date::test_date_format_day stderr ----
thread 'main' panicked at 'Stdout does not match regex:

', tests\common\util.rs:446:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_date::test_date_nano_seconds stdout ----
current_directory_resolved: 
run: D:\a\coreutils\coreutils\target\debug\coreutils.exe date +%N
---- test_date::test_date_nano_seconds stderr ----
thread 'main' panicked at 'Stdout does not match regex:

', tests\common\util.rs:446:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_date::test_date_rfc_3339 stdout ----
current_directory_resolved: 
run: D:\a\coreutils\coreutils\target\debug\coreutils.exe date --rfc-3339=ns
---- test_date::test_date_rfc_3339 stderr ----
thread 'main' panicked at 'Stdout does not match regex:

', tests\common\util.rs:446:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_date::test_date_format_day
    test_date::test_date_nano_seconds
    test_date::test_date_rfc_3339
```

**Review by sylvestre — CHANGES_REQUESTED** [review]
need test fixes

**Comment by rayan-mazouz** [comment]
closing as I don't have time to work on it

**Pull request closed without merging**